src/lib/components/table/table-column-settings/table-column-settings.component.spec.ts

```diff
@@ -0,0 +1,87 @@
+import {
+    ComponentFixture,
+    TestBed
+} from '@angular/core/testing';
+import { LocalizationModule } from 'angular-l10n';
+import Spy = jasmine.Spy;
+import {
+    TerraButtonComponent,
+    TooltipDirective
+} from '../../..';
+import {
+    MatDialog,
+    MatDialogModule
+} from '@angular/material/dialog';
+import { Router } from '@angular/router';
+import { MockRouter } from '../../../testing/mock-router';
+import { MatListModule } from '@angular/material/list';
+import { FormsModule } from '@angular/forms';
+import {
+    MatTableModule
+} from '@angular/material/table';
+import {
+    NgModule
+} from '@angular/core';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MatSelectModule } from '@angular/material/select';
+import { BrowserModule } from '@angular/platform-browser';
+import { TableSettingsComponent } from './table-settings.component';
+import { TableSettingsDialogComponent } from './dialog/table-settings-dialog.component';
+
+describe('TableSettingsComponent', () =>
+{
+    let component:TableSettingsComponent;
+    let fixture:ComponentFixture<TableSettingsComponent>;
+
+    beforeEach(() =>
+    {
+        TestBed.configureTestingModule(
+            {
+                declarations: [TableSettingsComponent],
+                imports:      [
+                    LocalizationModule.forRoot({}),
+                    FormsModule,
+                    BrowserAnimationsModule,
+                    MatTableModule,
+                    TestModule]
+            }
```

Comment:
Das Setup ist ganz schön aufwendig. Ich glaube das können wir noch etwas vereinfachen.

Reply:
Z.B. in dem wir MatDialog mocken und selbst providen. Dann brauchst du das ganze TestModule nicht mehr.

Reply:
Mat Components mocken ist ziemlich beschissen meiner Erfahrung nach

Reply:
Das ist keine Component, das ist ein Service. Das habe ich schonmal gemacht. Kann dir gerne zeigen was ich meine/wie das geht.